src/Domain/FileProcessors/FixerFileProcessor.php

```diff
@@ -58,19 +65,39 @@ public function processFile(SplFileInfo $splFileInfo): void
         }
 
         $oldContent = $splFileInfo->getContents();
-
         $tokens = Tokens::fromCode($oldContent);
+        $needFix = false;
         /** @var FixerDecorator $fixer */
         foreach ($this->fixers as $fixer) {
             $fixer->fix($splFileInfo, $tokens);
             if (! $tokens->isChanged()) {
                 continue;
             }
 
-            $fixer->addDiff($filePath, $this->differ->diff($oldContent, $tokens->generateCode()));
+            if ($this->fixEnabled === true) {
+                $needFix = true;
+                // Register diff will be applied
+                $fixer->registerFixedDiff($splFileInfo, $this->differ->diff($oldContent, $tokens->generateCode()));
+            }
+
+            if ($this->fixEnabled === false) {
```

Comment:
Is there a reason for this not to be an `else` statement?

Reply:
:sweat_smile: Yes ! Because I have the habit to work with phpmd that disallow `else` expression
https://phpmd.org/rules/cleancode.html#elseExpression 

But I have to admit it's not really clear here. I should rework this part :wink:

Reply:
Haha 😄 But yeah in this case here, it's not clear if it's because `fixEnabled` can be other values than bool, as you do a strict compare (which I totally prefer)